From 222ef3c534fc18f33c5fbf57c336392090273b1e Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Tue, 8 Apr 2014 15:22:13 +0200
Subject: [PATCH 53/56] EmailTest: Make it more like the agent

The agent commits the changes every 1 second, the test now does the same
thing. Previously it would test after each item, which was not very
practical and was useless for testing.
---
 src/pim/agent/tests/emailtest.cpp | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/src/pim/agent/tests/emailtest.cpp b/src/pim/agent/tests/emailtest.cpp
index 25103ab..5746975 100644
--- a/src/pim/agent/tests/emailtest.cpp
+++ b/src/pim/agent/tests/emailtest.cpp
@@ -44,6 +44,7 @@ private Q_SLOTS:
     void indexNextCollection();
     void itemReceived(const Akonadi::Item::List& item);
     void slotIndexed();
+    void slotCommitTimerElapsed();
 
 private:
     Akonadi::Collection::List m_collections;
@@ -52,6 +53,8 @@ private:
     QTime m_totalTime;
     int m_indexTime;
     int m_numEmails;
+
+    QTimer m_commitTimer;
 };
 
 int main(int argc, char** argv)
@@ -69,6 +72,10 @@ App::App(int& argc, char** argv, int flags)
 
 void App::main()
 {
+    m_commitTimer.setInterval(1000);
+    connect(&m_commitTimer, SIGNAL(timeout()), this, SLOT(slotCommitTimerElapsed()));
+    m_commitTimer.start();
+
     Akonadi::CollectionFetchJob* job = new Akonadi::CollectionFetchJob(Akonadi::Collection::root(),
                                                                        Akonadi::CollectionFetchJob::Recursive);
     connect(job, SIGNAL(finished(KJob*)), this, SLOT(slotRootCollectionsFetched(KJob*)));
@@ -124,8 +131,13 @@ void App::itemReceived(const Akonadi::Item::List& itemList)
 
     m_indexTime += timer.elapsed();
     m_numEmails += itemList.size();
+}
+
+void App::slotCommitTimerElapsed()
+{
+    QTime timer;
+    timer.start();
 
-    timer.restart();
     m_indexer.commit();
     m_indexTime += timer.elapsed();
 
-- 
1.9.1

